### PR TITLE
feature(v3): Cloud Dedicated write API: partial writes write good lines and reject bad li…

### LINKED ISSUE #5160: Write: partial writes write good lines and reject bad lines
See detail in https://github.com/influxdata/influxdb_iox/issues/5226#issuecomment-1716158763

Response:
`400`

```
{
  "code": "invalid",
  "line": 123,
  "message": "human-readable error message"
}
```
[Slack Message](https://influxdata.slack.com/archives/C04768675QD/p1696034786933909?thread_ts=1695822639.441929&cid=C04768675QD)

### PR DESCRIPTION
**Pull request by jstirnaman**
…nes #5160

**In draft until the CST change is completed**

- Documents the proposed partial writes feature for Cloud Dedicated:
  - `201`: partial write with response body describing rejected points
  - `204`:  the entire batch is written
- Mentions configurability of partial writes in CST, but still need information about how/who.
- Updates Troubleshoot writes and API docs.
- Links API compatibility docs to API reference, and vice versa.
- Fixes bad links in API docs.
- Removes rate limit and quota status codes from Cloud Dedicated.

Closes #5160

### CODE REVIEW

**Comment by jstirnaman** [comment]
@wiedld @jacobmarble PR for partial writes on CST

File: api-docs/cloud-dedicated/ref.yml [hunk 1]
```diff
@@ -1442,14 +1442,27 @@ paths:
...
+
+        The [`/write`](#operation/GetLegacyQuery) and [`/api/v2/write`](#operation/PostWrite) endpoints are functionally equivalent for writing data to InfluxDB Cloud Dedicated.
+
+        - Use the [`/write` endpoint](#operation/GetLegacyQuery) for InfluxDB v1 parameter compatibility.
+        - Use [`/api/v2/write` endpoint](#operation/PostWrite) for InfluxDB v2 parameter compatibility.
+
+        #### Rate limits
+
+        _Write_ rate limits apply.
+        For more information, see [limits and adjustable quotas](/influxdb/cloud-dedicated/admin/billing/limits/).
```
Comment (jacobmarble) [inline api-docs/cloud-dedicated/ref.yml:1465]: I didn't know we have write rate limits in Dedicated. TIL!

**Review by jacobmarble — APPROVED** [review]

File: api-docs/cloud-dedicated/ref.yml [hunk 2]
```diff
@@ -1576,25 +1589,43 @@ paths:
           - [Best practices for optimizing writes](/influxdb/cloud-dedicated/write-data/best-practices/optimize-writes/)
         required: true
       responses:
-        '204':
+        "201":
```
Comment (wiedld) [inline api-docs/cloud-dedicated/ref.yml:1592]: As of right now, the decisions is to have Dedicated be:
* 204 == everything written. regardless of partial-writes config.
* 201 == partial writes, partial errors. Only occurs in partial-write mode.

File: api-docs/cloud-dedicated/ref.yml (hunk 1 above)
Comment (jstirnaman) [reply api-docs/cloud-dedicated/ref.yml:1465]: > I didn't know we have write rate limits in Dedicated. TIL!

Oops

**Review by sanderson — APPROVED** [review]
👍🏻

File: content/influxdb/cloud-serverless/write-data/troubleshoot.md [hunk 3]
```diff
@@ -68,6 +79,37 @@ If you notice data is missing in your database, do the following:
...
+  "line": 2,
+  "message": "failed to parse line protocol:
+              errors encountered on line(s):
+              error parsing line 2 (1-based): Invalid measurement was provided
+              error parsing line 4 (1-based): Unable to parse timestamp value '123461000000000000000000000000'"
+}
+```
 
-Check for [field data type](/influxdb/cloud-serverless/reference/syntax/line-protocol/#data-types-and-format) differences between the rejected data point and points in the bucket that have the same measurement and day--for example, did you attempt to write `string` data to an `int` field?
+Check for [field data type](/influxdb/cloud-serverless/reference/syntax/line-protocol/#data-types-and-format) differences between the rejected data point and points within the same database and partition--for example, did you attempt to write `string` data to an `int` field?
```
Comment (jstirnaman) [inline content/influxdb/cloud-serverless/write-data/troubleshoot.md:115]:
```suggestion
Check for [field data type](/influxdb/cloud-serverless/reference/syntax/line-protocol/#data-types-and-format) differences between the rejected data point and points in the bucket that have the same measurement and day--for example, did you attempt to write `string` data to an `int` field?
```

File: api-docs/cloud-dedicated/ref.yml [hunk 4]
```diff
@@ -1442,14 +1442,29 @@ paths:
...
-        2. If successful, attempts to [ingest the data](/influxdb/cloud-dedicated/reference/internals/durability/#data-ingest); _error_ otherwise.
-        3.  If successful, responds with _success_ (HTTP `204` status code), acknowledging that the data is written and queryable; _error_ otherwise.
+        1. Validates the request.
+        2. If successful, attempts to [ingest data](/influxdb/cloud-dedicated/reference/internals/durability/#data-ingest) from the request body; otherwise, responds with an [error status](/influxdb/cloud-dedicated/write-data/troubleshoot/#review-http-status-codes).
+        3. Ingests or rejects data in the batch and returns one of the following HTTP status codes:
 
-        To ensure that InfluxDB Cloud handles writes in the order you request them,
-        wait for a success response (HTTP `2xx` status code) before you send the next request.
+           - `204 No Content`: all data in the batch is ingested
+           - `201 Created` (_If the cluster is configured to allow **partial writes**_): some points in the batch are ingested and queryable, and some points are rejected
```
Comment (sanderson) [inline api-docs/cloud-dedicated/ref.yml:1452]: As noted in Slack, I think this approach to documenting optional behavior is good. I think we just need to link to somewhere that provides information on how users can have their cluster configured for partial writes.

**Review by sanderson — APPROVED** [review]

**Pull request merged**